WordPress/gutenberg
# Font Library: show error if fetching collection fails

## What?
This PR adds error handling to rendering the font collection.  

## Why?
Fixes #54798

## How?
Throws an error if fetching the endpoint fails.

## Testing Instructions

1. Change [this line](https://github.com/WordPress/gutenberg/blob/626e1ef7be5b4cffea9d96fab92483210671a3ea/lib/experimental/fonts/font-library/font-library.php#L65) to something invalid
2. Open the Font Library > Install Fonts tab
3. Verify an error is shown

### Testing Instructions for Keyboard


## Screenshots or screencast 
Before | After
---- | ----
<img width="1624" alt="Screenshot 2023-09-28 at 3 47 39 PM" src="https://github.com/WordPress/gutenberg/assets/5375500/afa8d84b-ef3b-42ef-b1b2-1c8cc820fb3f"> | <img width="1624" alt="Screenshot 2023-09-28 at 3 39 04 PM" src="https://github.com/WordPress/gutenberg/assets/5375500/e3781a6d-fbe0-429f-9e71-0d9acde05333">
